agent/consul/state/intention.go

```diff
@@ -345,22 +379,40 @@ func (s *Store) IntentionMatch(ws memdb.WatchSet, args *structs.IntentionQueryMa
 
 // intentionMatchGetParams returns the tx.Get parameters to find all the
 // intentions for a certain entry.
-func (s *Store) intentionMatchGetParams(entry structs.IntentionMatchEntry) ([][]interface{}, error) {
-	// We always query for "*/*" so include that. If the namespace is a
-	// wildcard, then we're actually done.
-	result := make([][]interface{}, 0, 3)
-	result = append(result, []interface{}{"*", "*"})
-	if entry.Namespace == structs.IntentionWildcard {
-		return result, nil
-	}
+func (s *Store) intentionMatchGetParams(matchType structs.IntentionMatchType,
+	srcType structs.IntentionSourceType, entry structs.IntentionMatchEntry) ([][]interface{}, error) {
+
+	dstGetParams := func() [][]interface{} {
```

Comment:
Is this an inline function to make breaking out earlier easy?

If so you could refactor it into a `for {}` loop and replace the returns with breaks. It will be more performant than having to call a function.